Common/Field/src/MagneticField.cxx

```diff
@@ -558,18 +558,29 @@ MagFieldParam::BMap_t MagneticField::getFieldMapScale(float& l3, float& dip, boo
 
   MagFieldParam::BMap_t map = MagFieldParam::k5kG;
   float sclL3, sclDip;
+  float l3sav = l3, dipsav = dip;
 
   float l3Pol = l3 > 0 ? 1 : -1;
   float diPol = dip > 0 ? 1 : -1;
 
   l3 = TMath::Abs(l3);
   dip = TMath::Abs(dip);
+  static bool overrideL3 = std::getenv("O2_OVERRIDE_L3_CURRENT") != nullptr;
+  static bool overrideDIP = std::getenv("O2_OVERRIDE_DIPOLE_CURRENT") != nullptr;
+  static bool warnL3Done = false, warnDipDone = false, warnPolarityDone = false;
 
   if (TMath::Abs((sclDip = dip / diNominalCurrent) - 1.) > tolerance && !uniform) {
     if (dip <= zero) {
       sclDip = 0.; // some small current.. -> Dipole OFF
     } else {
-      LOG(fatal) << "MagneticField::createFieldMap: Wrong dipole current (" << dip << " A)!";
+      if (!overrideDIP) {
+        LOG(fatal) << "MagneticField::createFieldMap: Wrong dipole current (" << dipsav << " A)!";
+      } else {
+        if (!warnDipDone) {
```

Comment:
shouldn't this be a fatal as well? If it is an unsupported value, we don't have a field map for it, have we?

Reply:
We will not have good map but still reasonable scaling will be applied to existing maps. E.g one can emulate L3 Off / Dipole On field, will not be good for physics but e.g. for ED will work (better then 0/0).